packages/amplify-codegen/src/codegen-config/index.js

```diff
@@ -1,12 +1,69 @@
 const AmplifyCodeGenConfig = require('./AmplifyCodeGenConfig');
+const fs = require('fs-extra');
+const path = require('path');
 
 let config = null;
 
 function loadConfig(context, withoutInit = false) {
   if (!config) {
-    config = new AmplifyCodeGenConfig(context, withoutInit);
+    const projectPath = context.amplify.getEnvInfo().projectPath;
+    config = new AmplifyCodeGenConfig(projectPath, withoutInit);
   }
   return config;
 }
 
-module.exports = loadConfig;
+function getCodegenConfig(projectPath) {
+  if (!projectPath) {
+    throw new Error('Invalid projectPath provided to get codegen configuration');
+  }
+
+  if (!fs.existsSync(projectPath)) {
+    throw new Error(`Provided projectPath for getting the codegen configuration does not exist: ${projectPath}`);
+  }
+
+  if (!fs.existsSync(path.join(projectPath, AmplifyCodeGenConfig.configFileName))) {
+    throw new Error(`Cannot find the codegen configuration file at provided projectPath: ${projectPath}`);
+  }
+
+  const codegenConfig = new AmplifyCodeGenConfig(projectPath);
+  if (!codegenConfig) {
+    throw new Error(`Fetched codegen configuration is empty: ${codegenConfig}`);
+  }
+
+  const projects = codegenConfig.getProjects();
+  if (!projects.length > 0) {
+    throw new Error(`No projects were found in fetched codegen configuration: ${codegenConfig}`);
+  }
+
+  const cfg = projects[0];
+  const includeFiles = cfg.includes[0];
+
+  const opsGenDirectory = cfg.amplifyExtension.docsFilePath
+    ? cfg.amplifyExtension.docsFilePath
+    : path.dirname(path.dirname(includeFiles));
```

Comment:
Just curious on the case of docFilePath not being defined. Do we have a walkthrough case ending up in this situation? I look into the [code](https://github.com/aws-amplify/amplify-codegen/blob/f96e002435cc1c4c31d199cd4ffdca17f0856f59/packages/amplify-codegen/src/walkthrough/configure.js#L58) and do not find any unless the `.graphqlconfig` is manually edited.

Reply:
If it happens on manual edit only, I think we also need to check the `includeFiles` is in the correct form of glob expression ending up with `**/.*graphql`

Reply:
I kind of took the same logic we use for [statements generation](https://github.com/aws-amplify/amplify-codegen/blob/f96e002435cc1c4c31d199cd4ffdca17f0856f59/packages/amplify-codegen/src/commands/statements.js#L39) to compute the destination.